src/detect-engine.c

```diff
@@ -4118,6 +4122,7 @@ int DetectEngineReload(const SCInstance *suri)
         return -1;
     }
     SCLogDebug("set up new_de_ctx %p", new_de_ctx);
+    DatasetRemoveCopies();
```

Comment:
this needs to move until after `DetectEngineReloadThreads` for sure, probably until after `DetectEnginePruneFreeList` to avoid removing sets that are still being referenced by the old instance of the detection engine